xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiTradeService.java

```diff
@@ -0,0 +1,177 @@
+    CoingiPlaceLimitOrderRequest request =
+        new CoingiPlaceLimitOrderRequest()
+            .setCurrencyPair(CoingiAdapters.adaptCurrency(order.getCurrencyPair()))
+            .setOrderType(order.getType().equals(BID) ? 0 : 1)
+            .setPrice(order.getLimitPrice())
+            .setVolume(order.getRemainingAmount());
+
+    String response;
+    try {
+      response = placeCoingiLimitOrder(request).getResult();
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+
+    return response;
+  }
+
+  @Override
+  public boolean cancelOrder(String orderId) throws IOException {
+    // if it doesn't return an error and CoingiOrder is returned, then the cancellation is
+    // successful
+    CoingiCancelOrderRequest request = new CoingiCancelOrderRequest();
+    request.setOrderId(orderId);
+    CoingiOrder order;
+    try {
+      order = cancelCoingiOrder(request);
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+
+    return order != null;
+  }
+
+  @Override
+  public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
+    if (orderParams instanceof CancelOrderByIdParams) {
+      return cancelOrder(((CancelOrderByIdParams) orderParams).getOrderId());
+    } else {
+      return false;
+    }
+  }
+
+  /** Required parameter types: {@link TradeHistoryParamPaging#getPageLength()} */
+  @Override
+  public UserTrades getTradeHistory(TradeHistoryParams p) throws IOException {
+    CoingiOrdersList orderList;
+
+    if (p instanceof CoingiTradeHistoryParams) {
+      CoingiTradeHistoryParams params = (CoingiTradeHistoryParams) p;
+
+      CoingiGetOrderHistoryRequest request = new CoingiGetOrderHistoryRequest();
+
+      if (params.getCurrencyPair() != null) request.setCurrencyPair(params.getCurrencyPair());
+
+      request.setPageNumber(params.getPageNumber());
+      request.setPageSize(params.getPageSize());
+
+      if (params.getStatus() != null) request.setStatus(params.getStatus());
+
+      if (params.getType() != null) request.setType(params.getType());
+
+      try {
+        orderList = getCoingiOrderHistory(request);
+      } catch (CoingiException e) {
+        throw CoingiErrorAdapter.adapt(e);
+      }
+
+      return CoingiAdapters.adaptTradeHistory(orderList);
+    }
+
+    return null;
+  }
+
+  @Override
+  public TradeHistoryParams createTradeHistoryParams() {
+    return new CoingiTradeHistoryParams(null, 1, 30, null, null);
+  }
+
+  @Override
+  public OpenOrdersParams createOpenOrdersParams() {
+    return new DefaultOpenOrdersParamCurrencyPair();
+  }
+
+  @Override
+  public Collection<Order> getOrder(String... orderIds) throws IOException {
+    Collection<Order> orders = new ArrayList<>();
+    for (String orderId : orderIds) {
+      CoingiGetOrderRequest request = new CoingiGetOrderRequest().setOrderId(orderId);
+      CoingiOrder coingiOrder;
+      try {
+        coingiOrder = getCoingiOrder(request);
```

Comment:
the try / catch CoingiException  hit not applied

Reply:
done, thanks for spotting this